### PR TITLE
[WIP]Paper: UserStudy

### PR DESCRIPTION
**Pull request by beyond2void**
Sketched a structure of the part User Study.
Question: 
- Do you think we should rename the part?
- Or maybe we could put this part as subsection under some section.(Idk)
- What other points should be mentioned in the part "User Study"?

### CODE REVIEW

File: paper/user_study.tex [hunk 1]
```diff
@@ -0,0 +1,66 @@
+Our user study is mainly based on the qualitative research.
```
Comment (PaulBredl) [inline paper/user_study.tex:1]: I would leave out this sentence
```suggestion
```

File: paper/user_study.tex [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+We conduct several interviews with the professors and students as potential customers.
```
Comment (PaulBredl) [inline paper/user_study.tex:2]:
```suggestion
We conduct several interviews with professors and students as potential customers.
```

File: paper/user_study.tex [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+We collect feedbacks from the customers, improve user experiences and fix bugs according to the feedback.
```
Comment (PaulBredl) [inline paper/user_study.tex:3]:
```suggestion
We collect feedback from the customers to improve user experience and fix bugs according to the feedback.
```

File: paper/user_study.tex [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+We also use the survey to get interviewees' opinions during the user study.
```
Comment (PaulBredl) [inline paper/user_study.tex:4]: What do you mean with that sentence? A survey is more the thing we did for demographic quesions
```suggestion
We also use a survey to get interviewees' opinions during the user study.
```

File: paper/user_study.tex [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+\subsection{Methodology}\label{subsec:methodology}
+
+\paragraph{Potential Users}
+We look for potential users to conduct the interviews.
+There are two basic standards we follow to find a potential user:
+\begin{itemize}
+    \item Professors and students who are interested in our application.
+    \item People who frequently use \cfgfiles and schema languages
+\end{itemize}
+Because each interviewee has different prior knowledge of JSON and JSON schema, we improve our interview questions and the example data also after the interview.
```
Comment (PaulBredl) [inline paper/user_study.tex:15]: Do we? I think not really?

File: paper/user_study.tex [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+So that our interview is more compatible for users.
+
+\paragraph{Interview Questions}
+We prepare our interview questions based on a mockup JSON schema and \cfgfiles.
+It is about a simple and understandable autonomous vehicle schema.
```
Comment (PaulBredl) [inline paper/user_study.tex:20]: It's not about a autonomous vehicle schema. It's about a (made-up) simulation software that simulates a self-driving car.

File: paper/user_study.tex [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+\begin{itemize}
+    \item Setup Question: Where to open the application and set up schema and \cfgfiles.
+    \item Basic Question: Goal is to know about the GUI of application
+    \item Modifying the configuration file: Solving problems of \cfgfiles inside the \textbf{file editor}.
+    \item Modifying the schema: Solving problems of JSON schema inside the \textbf{schema editor}.
+\end{itemize}
+
+\paragraph{Interview Process}
+Commonly, we conduct our interview with one interviewee each time.
+The interview lasts often within one hour.
```
Comment (PaulBredl) [inline paper/user_study.tex:31]: This is not correct grammar

File: paper/user_study.tex [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+\paragraph{Survey}
+After the interview, we ask each interviewee to finish a survey(follow-up questions).
+This survey will cover the following points:
+\begin{itemize}
+    \item Occupation of the interviewee.
+    \item Experiences in software engineering,
+    \item Frequency of working with JSON/YAML files.
+    \item Domains in which people use with JSON/YAML files.
+    \item Feedback about the application after interview.
```
Comment (PaulBredl) [inline paper/user_study.tex:46]: Did we add that? I thought we dont have that

File: paper/user_study.tex [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+\end{itemize}
+Based on the survey, we could get many critical points which we can use in the future.
+We analyse the results which we get from the survey and improve our application afterwards.
+Details found in ~\ref{improvements}
+
+\paragraph{Improvements}\label{improvements}
+After getting results from the interview and survey, we have an internal meeting inside our team and discuss which features should be added to our application.
```
Comment (PaulBredl) [inline paper/user_study.tex:53]: This is not important

**Review by PaulBredl — CHANGES_REQUESTED** [review]

File: paper/user_study.tex (hunk 5 above)
Comment (beyond2void) [reply paper/user_study.tex:15]: We simplified our questions once, but It's not important to write this here I think.
So I'll delete it.

File: paper/user_study.tex (hunk 7 above)
Comment (beyond2void) [reply paper/user_study.tex:31]: How do you think I should edit here?

File: paper/user_study.tex (hunk 8 above)
Comment (beyond2void) [reply paper/user_study.tex:46]: Ah. I wrote it based on our survey. Did we change a lot afterwards?

File: paper/user_study.tex [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+    \item Experiences in software engineering,
+    \item Frequency of working with JSON/YAML files.
+    \item Domains in which people use with JSON/YAML files.
+    \item Feedback about the application after interview.
+\end{itemize}
+Based on the survey, we could get many feedbacks which we can use in the future.
+We analyse the results which we get from the survey and improve our application afterward.
+
+\subsection{Results}
+% todo: show specific cases of user study. Use table of feedbacks and resolutions
```
Comment (PaulBredl) [inline paper/user_study.tex:49]: Add table of correctness of the participants

**Review by PaulBredl — APPROVED** [review]
for now we can merge that @beyond2void

**Pull request merged**